### PR TITLE
fix: make it a compile time error when the actor bundle size is invalid

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- Make it a compile time error when the actor bundle size is invalid
- Add a `cargo publish --dry-run` CI job to capture errors like assets are invalid, assets are not properly included, and temporary crate patch, etc. that blocks cargo publishing.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by aatifsyed — APPROVED** [review]

File: .github/workflows/forest.yml [hunk 1]
```diff
@@ -74,6 +74,28 @@ jobs:
             ~/.cargo/bin/forest*
           if-no-files-found: error
 
+  cargo-publish-dry-run:
+    name: cargo publish --dry-run
```
Comment (LesnyRumcajs) [inline .github/workflows/forest.yml:78]: The additional name isn't really more helpful so we might as well drop it.
```suggestion
```

File: .github/workflows/forest.yml [hunk 2]
```diff
@@ -74,6 +74,28 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      # To help investigate transient test failures
+      - run: lscpu
+      # find the nearest S3 space for storing cache files
+      - name: Show IP
+        run: curl ifconfig.me
+        continue-on-error: true
```
Comment (LesnyRumcajs) [inline .github/workflows/forest.yml:86]: Why do we need it in this job exactly? If there isn't anything special about it, let's either have it everywhere (ideally via a composite action to reduce code duplication) or only in critical places.

File: src/daemon/bundle.rs [hunk 3]
```diff
@@ -5,11 +5,15 @@ use anyhow::Context;
...
+use static_assertions::const_assert_eq;
 
 pub async fn load_actor_bundles(db: &impl Blockstore) -> anyhow::Result<Vec<Cid>> {
     const ERROR_MESSAGE: &str = "Actor bundles assets are not properly downloaded, make sure git-lfs is installed and run `git lfs pull` again. See <https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md>";
+
     const ACTOR_BUNDLES_CAR_ZST: &[u8] = include_bytes!("../../assets/actor_bundles.car.zst");
-    assert!(ACTOR_BUNDLES_CAR_ZST.len() > 1024 * 1024, "{ERROR_MESSAGE}");
+    // Check bundle size at compile time, see `ERROR_MESSAGE` for details.
+    // Note: make sure the bundle size is updated with the bundle file.
+    const_assert_eq!(ACTOR_BUNDLES_CAR_ZST.len(), 2438387);
```
Comment (LesnyRumcajs) [inline src/daemon/bundle.rs:16]: For the sake of completeness, it'd be helpful to provide instructions on how to regenerate the bundle. This way, a potential network upgrade implementor would be less frustrated by this sudden obstacle.

File: .github/workflows/forest.yml (hunk 2 above)
Comment (hanabi1224) [reply .github/workflows/forest.yml:85]: Removed

File: src/daemon/bundle.rs (hunk 3 above)
Comment (hanabi1224) [reply src/daemon/bundle.rs:16]: I'm working on a `forest-tool` subcommand that regenerates the bundle, and will update instructions in STATE_MIGRATION.md altogether

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**